db/blip_sync_context.go

```diff
@@ -655,3 +657,24 @@ func toHistory(revisions Revisions, knownRevs map[string]bool, maxHistory int) [
 	}
 	return history
 }
+
+// Remembers a docID that doesn't exist in the bucket at the time handleProposeChanges ran.
+func (bsc *BlipSyncContext) notePendingInsertion(docID string) {
+	bsc.allowedAttachmentsLock.Lock() // TODO: Rename this lock?
+	defer bsc.allowedAttachmentsLock.Unlock()
```

Comment:
I don't think this has to be allowedAttachmentsLock as above, this is collection specific locking